test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildIntrospectionTest.cs

```diff
@@ -19,5 +19,37 @@ public class BuildIntrospectionTest : MSBuildIntegrationTestBase
             Assert.BuildOutputContainsLine(result, $"UpToDateCheckInput: {Path.Combine("Views", "Home", "Index.cshtml")}");
             Assert.BuildOutputContainsLine(result, $"UpToDateCheckInput: {Path.Combine("Views", "_ViewStart.cshtml")}");
         }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task RazorSdk_SetsPreserveCompilationContextForProjectsWithCshtmlFiles()
+        {
+            var result = await DotnetMSBuild("Build", "/t:_IntrospectPreserveCompilationContext");
+
+            Assert.BuildPassed(result);
+            Assert.BuildOutputContainsLine(result, "PreserveCompilationContext: true");
```

Comment:
Is it possible to verify a build output rather than text?

Relying on introspection is fragile - if we set the property after its been used we wouldn't know it broke. We should use it as a last resort?

Reply:
Sure. I can make sure the output file has the requisite `CompilationLibrary` instances Mvc would care about